sonarts-core/src/rules/argumentsOrderRule.ts

```diff
@@ -0,0 +1,137 @@
+      const hasProblem = argumentNames.some((argumentName, argumentIndex) => {
+        if (!argumentName) {
+          return false;
+        }
+        return this.isArgumentSwapped(argumentName, argumentIndex, argumentNames, parameterNames, node);
+      });
+
+      if (hasProblem) {
+        this.raiseIssue(node, declaration);
+      }
+    }
+    super.visitCallExpression(node);
+  }
+
+  isArgumentSwapped(
+    argumentName: string,
+    argumentIndex: number,
+    argumentNames: (string | undefined)[],
+    parameterNames: ts.Symbol[],
+    node: ts.CallExpression,
+  ) {
+    const parameterIndex = parameterNames.findIndex(param => param.name === argumentName);
+    if (parameterIndex > -1 && parameterIndex !== argumentIndex) {
+      const anotherArgument = argumentNames[parameterIndex];
+      if (parameterNames[argumentIndex].name === anotherArgument) {
+        const firstArgumentType = this.getTypeAsString(node.arguments[argumentIndex]);
+        const secondArgumentType = this.getTypeAsString(node.arguments[parameterIndex]);
+
+        if (firstArgumentType === secondArgumentType) {
+          return true;
+        }
+      }
+    }
+
+    return false;
+  }
+
+  getTypeAsString(expr: ts.Expression) {
+    const typeChecker = this.program.getTypeChecker();
+    return typeChecker.typeToString(typeChecker.getBaseTypeOfLiteralType(typeChecker.getTypeAtLocation(expr)));
+  }
+
+  raiseIssue(node: ts.CallExpression, declaration?: ts.SignatureDeclaration | ts.JSDocSignature) {
+    const issue = this.addIssueAtLocation(
+      new IssueLocation(node.arguments.pos, node.arguments.end, node.getSourceFile(), Rule.MESSAGE),
+    );
+    if (
+      declaration &&
+      (declaration.parameters as ts.TextRange).pos &&
+      declaration.getSourceFile() === node.getSourceFile()
+    ) {
+      const textRange: ts.TextRange = declaration.parameters as ts.TextRange;
+      issue.addSecondaryLocation(
+        new IssueLocation(textRange.pos, textRange.end, node.getSourceFile(), "Formal parameters"),
+      );
+    }
+  }
+
+  getArgumentNames(node: ts.CallExpression) {
+    const argumentNames: (string | undefined)[] = [];
+    node.arguments.forEach(arg => {
+      if (isIdentifier(arg)) {
+        argumentNames.push(arg.getText());
+      } else {
+        argumentNames.push(undefined);
+      }
+    });
+
+    return argumentNames;
+  }
+
+  getSignature(node: ts.CallExpression) {
+    const parameterNames: ts.Symbol[] = [];
+    const calleeType = this.program.getTypeChecker().getTypeAtLocation(node.expression);
+    const callSignatures = calleeType.getCallSignatures();
+    if (callSignatures.length === 1) {
+      const callSignature = callSignatures[0];
+      callSignature.parameters.forEach(parameterSymbol => {
+        parameterNames.push(parameterSymbol);
+      });
+
+      return { parameterNames, declaration: callSignature.declaration };
+    }
+
+    return { parameterNames: undefined, declaration: undefined };
```

Comment:
This line is not covered by unit tests, we should test when we don't have a call signature (and when callSignatures.length > 1)